nixos/modules/services/web-apps/kimai.nix

```diff
@@ -0,0 +1,218 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  userCfg = config.services.kimai;
+  kimaiDrv = pkgs.kimai.override {
+      consoleCmd = userCfg.consoleCmd;
+      databaseUrl = userCfg.databaseUrl;
+      mailerFrom = userCfg.mailerFrom;
+      mailerUrl = userCfg.mailerUrl;
+      appSecret = userCfg.appSecret;
+      corsAllowOrigin = userCfg.corsAllowOrigin;
+    };
+  kimaiName = "kimai";
+  kimaiDataDir = "${kimaiDrv}/public";
+in
+{
+  options.services.kimai = {
+    enable = mkEnableOption (lib.mdDoc "Kimai web client");
+    consoleCmd = mkOption {
+      type = types.str;
+      default = "kimai-console";
+      description = lib.mdDoc ''
+        The name of the binary for Kimai. This would be the binary that can be called
+        for the Kimai commands, usually shown as `/bin/console` in the docs, but renamed in
+        this package to `kimai-console` for clarity.
+
+        - https://www.kimai.org/documentation/commands.html
+
+        This value becomes the name of the file, so don't use an invalid file name.
+      '';
+    };
+    enableDatabase = mkOption {
+      type = types.bool;
+      default = true;
+      description =  lib.mdDoc ''
+        Whether to configure mysql database for you, or not.
+        If you disabled this option then you will need to adjust the `databaseUrl` option
+        to fit your needs.
+      '';
+    };
+    mysqlPackage = mkPackageOptionMD pkgs "mysql80" {
+      extraDescription =  lib.mdDoc ''
+          Mysql package to use.
+          Note: changing this may require changing the `databaseUrl` option too.
+        '';
+      };
+    databaseUrl = mkOption {
+      type = types.str;
+      default = "mysql://kimai:@127.0.0.1:3306/kimai_db?charset=utf8mb4&serverVersion=8.0.34";
+      description = lib.mdDoc ''
+        The DATABASE_URL env var from KIMAI:
+        https://github.com/kimai/kimai/blob/main/.env.dist#L22C1-L22C13
+      '';
+    };
+    mailerFrom = mkOption {
+      type = types.str;
+      default = "kimai@example.com";
+      description = lib.mdDoc ''
+        The MAILER_FROM env var from KIMAI:
+        https://github.com/kimai/kimai/blob/main/.env.dist#L28
+      '';
+    };
+    mailerUrl = mkOption {
+      type = types.str;
+      default = "null://null";
+      description = lib.mdDoc ''
+        The MAILER_URL env var from KIMAI:
+        https://github.com/kimai/kimai/blob/main/.env.dist#L30
+      '';
+    };
+    appSecret = mkOption {
+      type = types.str;
+      default = "change_this_to_something_unique";
```

Comment:
This should be read from a file to be secure

Reply:
@SuperSandro2000 
Is it considered safer to ask for a file path instead of a string? Because I have not seen that in a lot of other packages.